rllib/offline/offline_data.py

```diff
@@ -0,0 +1,152 @@
+import functools
+import logging
+import numpy as np
+from pathlib import Path
+import ray
+from typing import Any, Dict, List, Union
+
+from ray.rllib.algorithms.algorithm_config import AlgorithmConfig
+from ray.rllib.core.columns import Columns
+from ray.rllib.env.single_agent_episode import SingleAgentEpisode
+from ray.rllib.utils.compression import unpack_if_needed
+from ray.rllib.utils.typing import EpisodeType
+
+logger = logging.getLogger(__name__)
+
+# TODO (simon): Implement schema mapping for users, i.e. user define
+# which row name to map to which default schema name below.
+SCHEMA = [
+    Columns.EPS_ID,
+    Columns.AGENT_ID,
+    Columns.MODULE_ID,
+    Columns.OBS,
+    Columns.ACTIONS,
+    Columns.REWARDS,
+    Columns.INFOS,
+    Columns.NEXT_OBS,
+    Columns.TERMINATEDS,
+    Columns.TRUNCATEDS,
+    Columns.T,
+    # TODO (simon): Add remove as soon as we are new stack only.
+    "agent_index",
+    "dones",
+    "unroll_id",
+]
+
+
+class OfflineData:
+    def __init__(self, config: Union[Dict[str, Any], AlgorithmConfig]):
+
+        self.config = config
+        self.is_multi_agent = (
```

Comment:
Let's enforce `self.config` to be an AlgorithmConfig.

This is perfectly aligned then with
* EnvRunnerGroup (takes `AlgorithmConfig` in c'tor) -> has n EnvRunners (take AlgorithmConfig in c'tor)
* LearnerGroup (takes `AlgorithmConfig` in c'tor) -> has m Learners (take AlgorithmConfig in c'tor)

Reply:
Alright. Let's do it! Makes a lot of things easier - only testing not. My initial thought was: so users can use this class outside of RLlib. But to be honest they can use their own ray.data.Dataset.

Reply:
Gotcha. But they can still use these components outside RLlib.
I don't think it's a problem doing:
```
from ray.rllib.algorithm.algorithm_config import AlgorithmConfig
from ray.rllib.env.single_agent_env_runner import SingleAgentEnvRunner

config = AlgorithmConfig.environment("CartPole-v1").rl_module(...)

my_env_runner = SingleAgentEnvRunner(config)
some_episodes = my_env_runner.sample()
```

I think that's still super neat and nimble to work with. We might want to rename - at some point in the future - `AlgorithmConfig` into `RLConfig`, something more generic, so users don't get scared of it.

Reply:
Or better: `RLlibConfig`